README.md

```diff
@@ -22,21 +22,529 @@ A Spezi framework that provides a common set of SwiftUI views and related functi
+        }
+    }
+}
+```
+
+### SpeziValidation - Input Validation
+
+Add real-time validation to forms:
+
+```swift
+import SpeziValidation
+import SwiftUI
+
+struct SignupView: View {
+    @State private var email = ""
+    @State private var password = ""
+    @State private var username = ""
+    
+    @ValidationState private var validation
+    
+    var body: some View {
+        Form {
+            Section("Account Details") {
+                VerifiableTextField("Email", text: $email)
+                    .validate(input: email, rules: .minimalEmail)
+                    .textInputAutocapitalization(.never)
+                    .autocorrectionDisabled()
+                
+                VerifiableTextField("Password", text: $password, type: .secure)
+                    .validate(input: password, rules: .minimalPassword)
+                
+                VerifiableTextField("Username", text: $username)
+                    .validate(input: username, rules: .nonEmpty)
+            } footer: {
+                Text("Choose a unique username")
+            }
+        }
+        .receiveValidation(in: $validation)
+        .toolbar {
+            ToolbarItem(placement: .primaryAction) {
+                Button("Create Account") {
+                    guard validation.validateSubviews() else {
+                        return
+                    }
+                    // Create account
+                }
+                .disabled(!validation.allInputValid)
+            }
+        }
+    }
+}
+```
+
+#### Custom Validation Rules
+
+Create custom validation logic:
+
+```swift
+import SpeziValidation
+import SwiftUI
+
+extension ValidationRule {
+    static let strongPassword = ValidationRule(
+        rule: { password in
+            password.count >= 8 && 
+            password.contains { $0.isUppercase } &&
+            password.contains { $0.isLowercase } &&
+            password.contains { $0.isNumber }
+        },
+        message: "Password must be at least 8 characters with uppercase, lowercase, and numbers"
+    )
+}
+
+struct SecureSignupView: View {
+    @State private var password = ""
+    
+    var body: some View {
+        Form {
+            VerifiableTextField("Password", text: $password, type: .secure)
+                .validate(input: password, rules: .strongPassword)
+        }
+    }
+}
+```
+
+#### Validation with Multiple Rules
+
+Combine multiple validation rules:
+
+```swift
+import SpeziValidation
+import SwiftUI
+
+struct ContactForm: View {
+    @State private var fullName = ""
+    @State private var email = ""
+    
+    var body: some View {
+        Form {
+            VerifiableTextField("Full Name", text: $fullName)
+                .validate(input: fullName, rules: [.nonEmpty, .unicodeLettersOnly])
+            
+            VerifiableTextField("Email", text: $email)
+                .validate(input: email, rules: .minimalEmail)
+        }
+    }
+}
+```
+
+## Additional Components and Utilities
```

Comment:
Idea: What do you think about a compact table with the name + a screenshot for all these components from here on that link to the line documentation to we can quickly jump to them rather than all the code examples here? Can also be applicable to some components above, maybe that's generally a nice format for the README?

The image + code example can then also be added to the inline-documentation. Will help the discovery but also doesn't make the README too long.